ports/esp32/modnetwork.c

```diff
@@ -398,7 +398,26 @@ STATIC mp_obj_t esp_status(size_t n_args, const mp_obj_t *args) {
             }
             return list;
         }
+        case (uintptr_t)MP_OBJ_NEW_QSTR(MP_QSTR_rssi): {
+            ESP_LOGD("wifi-status", "RSSI call");
+            // return signal of AP, only in STA mode
+            require_if(args[0], WIFI_IF_STA);
 
+            wifi_mode_t mode;
+            ESP_EXCEPTIONS(esp_wifi_get_mode(&mode));
+            if ((mode & WIFI_MODE_STA) == 0) {
+                nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "STA must be active"));
```

Comment:
I don't think this check is necessary.  The `require_if()` call above already checks that the interface is STA, and the `esp_wifi_sta_get_ap_info()` call below will probably return an error if the STA is not active.  It's best to keep things simple and minimal because in vast majority of uses of this function there will be no error.  The doesn't need to be redundant error checking.

Reply:
require_if only checks that interface is STA and not AP. Does not care if it's active or not. I spent almost 1 hour about solve "why wifi.connect()" returns error 0x0005 (which mean by SDK not enough power) just because I forgot do active(1)... Meanwhile I changed USB cables, got powered USB HUB. If there will be such check for "active" It will save my time and headache.

Reply:
Ok, but that would be a separate issue (connect() not raising an exception if it's not active).  Here, it should be that esp_wifi_sta_get_ap_info() will fail if the STA is not active.

Reply:
I mention that "connect" as example why do more logs or make logs be somehow "accurate". Anyway this error 0x0005 must be issue in SDK or in HW chip, because I did not found documented such error code (IDF, PDFs...). Also if you will do active(1) and then active(0) it will start report valid error code "Wifi not started". But for this I will do separated issue / pr.